### PR TITLE
[SW-1289] Move shared params to H2OCommonParams

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala [hunk 1]
```diff
@@ -433,14 +428,11 @@ trait H2OGridSearchParams extends H2OCommonParams with DeprecatableParams {
     strategy -> HyperSpaceSearchCriteria.Strategy.Cartesian,
     maxRuntimeSecs -> 0,
     maxModels -> 0,
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala]: Shouldn't ```allStringColumnsToCategorical``` and ```columnsToCategorical``` be removed as well?

File: py/pysparkling/ml/params.py [hunk 2]
```diff
@@ -48,21 +77,36 @@ def setWeightCol(self, value):
...
+    def setNfolds(self, value):
+        assert_is_type(value, int)
+        return self._set(nfolds=value)
+
+    def setAllStringColumnsToCategorical(self, value):
+        assert_is_type(value, bool)
+        return self._set(allStringColumnsToCategorical=value)
+
+    def setColumnsToCategorical(self, value):
+        assert_is_type(value, [str])
```
Comment (mn-mikke) [inline py/pysparkling/ml/params.py]: Shouldn't we also expose an overloaded method with a variable number of parameters?

Comment (jakubhava) [reply py/pysparkling/ml/params.py]: done

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala]: Done

**Pull request merged**